lua/entities/gmod_wire_dhdd.lua

```diff
@@ -14,11 +14,33 @@ function ENT:Initialize()
 	self.Outputs = WireLib.CreateOutputs( self, { "Memory [ARRAY]", "Size" } )
 	self.Inputs = WireLib.CreateInputs( self, { "Data [ARRAY]", "Clear", "AllowWrite" } )
 
-	self.Memory = {}
+	self:SetMemory()
 	self.ROM = false
 	self.AllowWrite = true
+end
+
+function ENT:SetMemory(Memory)
```

Comment:
I've just tested it and this should work fine creating the proxy table only once in `Initialize`, before `self.Memory` is even set.